### ISSUE TITLE
Support `fixup` workflow

### ISSUE THREAD

**Issue by Mirko-von-Leipzig** [extrawurst/gitui#2019]
**Is your feature request related to a problem? Please describe.**
I'd love a shortcut for `git commit --fixup <SHA>` from within gitui. [Here](https://dev.to/koffeinfrei/the-git-fixup-workflow-386d) is an article describing a common use case for it, but I'll give a short summary.

What this does is create a new commit with `fixup! <target <SHA> commit message>`. You can then run `git rebase -i --autosquash` to squash the fixup commits into the target commits. This allows you to defer fixes/ammends to previous commits without messing with git history.

A nice use case is when a PR is under review. It let's you mark changes to existing commits without the reviewers losing track of what's been reviewed (as you would if you did `git commit --amend` or a rebase). Once the PR is approved you neaten the commit history using `git rebase -i --autosquash`.

**Describe the solution you'd like**
I'd like a new shortcut on the commit panel to trigger a `git commit --fixup <SHA>`. e.g. `ctrl-f`. This would bring up a view similar to the `logs` panel which would allow choosing the commit to fixup.

**Describe alternatives you've considered**
Current workflow is:
1. Stage changes 
2. Find the commit in logs view, copy the commit hash
3. Exit gitui
4. Manually `git commit --fixup <PASTE>`
5. Re-enter gitui, continue

**Additional context**
This was previously requested in #829 which was closed as completed - I think incorrectly as I [commented](https://github.com/extrawurst/gitui/issues/829#issuecomment-1895881991).

I've created a new issue instead of attempting to resurrect that one.

**Comment by extrawurst** [comment]
see https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---fixupamendrewordltcommitgt for all dirty details and options 😅

**Comment by danriedl** [comment]
heck yeah, +1 for amend.

**Comment by wpcarro** [comment]
Whatever Magit's doing in Emacs for its "instant fixup" appears to "just work" (IOW... DWIM)

btw - ty @extrawurst for making a sweet tool :)

**Comment by Chinchzilla** [comment]
+1 this is a feature I miss the most after coming from LazyGit

**Comment by Dotrar** [comment]
Hey all, just chiming in: if some of you don't know about it, there is `git-absorb` which is written in rust. this does `--fixup` for you automatically, which is handy. 

I had a quick look around, and it seems to be using `git2` under the hood, so i'd say it should be possible to have in gitui as well.

The workflow I'd picture is:
* stage some files
* go to log (2)
* if there's staged files, then [f] is available to just commit as a fixup. pop up the editor to add a message as well. 


You can kinda already do this with what we've got:
* stage some files
* go to log (2)
* [y] to copy the target commit sha, [ret] to remove popup (would love to remove this popup tbh)
* then [1] [c] and just type "fixup! <paste><ret><ret> message".

It's trivial to implement; I'm not a rust programmer but i could give it a go..

**Comment by Dotrar** [comment]
I tried adding a PR as to the above, but it got knocked back a bit.

I think we're wanting a "interactive rebase" instead. We're still not sure about the UX of this feature.

**Comment by loserMcloser** [comment]
+1 for having a `fixup` feature.

From the comments to #2505:

> For fixups, it's best to do it the manual method of copying the sha and having a git message template

Hard disagree. Having
````
fixup! <long-commit-hash>
````
in your commit history is *much* less readable than having
````
fixup! <commit-subject-line>
````
which is what `commit --fixup` does by default on the CLI.

**Comment by Dotrar** [comment]
I agree. That could be a small modification to my pr.

I think we need to explain the UX of `fixup!` - it seemed like it was unintuitive to use the log to find the target commit. But I don't know where else it would come from.

I didn't have much room to argue and I don't know enough rust to make it a easy-to-merge change. 

(I'm now working on my own tool to do this instead (In python, which I'm a bit better at))
